legacy/src/main/java/org/opensearch/sql/legacy/pit/PointInTimeHandlerImpl.java

```diff
@@ -37,31 +39,62 @@ public PointInTimeHandlerImpl(Client client, String[] indices) {
     this.indices = indices;
   }
 
-  /** Create PIT for given indices */
+  /**
+   * Constructor for class
+   *
+   * @param client OpenSearch client
+   * @param pitId Point In Time ID
+   */
+  public PointInTimeHandlerImpl(Client client, String pitId) {
+    this.client = client;
+    this.pitId = pitId;
+  }
+
+  /**
+   * Create PIT for given indices
+   *
+   * @return Point In Time creation status
+   */
   @Override
-  public void create() {
+  public boolean create() {
+    CompletableFuture<Boolean> future = new CompletableFuture<>();
```

Comment:
This is not required. The client itself provides this.

https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/client/support/AbstractClient.java#L465
Please copy the code here:
https://github.com/opensearch-project/OpenSearch/blob/main/server/src/internalClusterTest/java/org/opensearch/search/slice/SearchSliceIT.java#L177

Reply:
Nice